### DISCUSSION TITLE
change default targets console output color?

### DISCUSSION THREAD

**Discussion by gorkang** [ropensci/targets#856] [Help] [answered]
Hi there!

TLDR: The targets output default color is red, which makes custom errors/warnings hard to see. Changing to blue could be a nice compromise? Alternatively, something like `options(targets.default_text_color = "white")` could work? Examples below.

---  

When targets is running in interactive mode, all the console output is in red. This makes warnings or errors in our custom functions really hard to find. 

For example: 

![Screenshot from 2022-06-16 12-34-30](https://user-images.githubusercontent.com/6491790/174062014-1903083c-8896-4f1e-8a0f-a8abc0f5a1d0.png)

Compare the above with a white text color for the targets outputs. This makes warnings/errors much easier to find:
![Screenshot from 2022-06-16 12-35-38](https://user-images.githubusercontent.com/6491790/174062892-f676f0d1-94d1-4c33-9116-15bba25e0a14.png)

I guess is the issue with using white comes when the console background is also white:
![Screenshot from 2022-06-16 12-48-46](https://user-images.githubusercontent.com/6491790/174063423-6fbfd972-ee08-4536-afee-1971a047e2ee.png)

So, what about blue? Seems good enough with white/black backgrounds:  

![both](https://user-images.githubusercontent.com/6491790/174064145-1873207b-fa89-437e-9762-306e705e5fe6.png)

And decent enough with bluish backgrounds:

![Screenshot from 2022-06-16 12-46-16](https://user-images.githubusercontent.com/6491790/174065032-06ae54ba-125b-4007-acb6-5c96e9645806.png)

To do this, I added `cli::col_blue()` to the msg var in the following functions in utils_cli.R:

```
cli_blue_bullet <- function(msg) {
  symbol <- cli::col_blue(cli::symbol$bullet)
  msg <- cli::col_blue(paste(symbol, msg))
  message(msg)
}

cli_green_bullet <- function(msg) {
  symbol <- cli::col_green(cli::symbol$bullet)
  msg <- cli::col_blue(paste(symbol, msg))
  message(msg)
}

cli_green_check <- function(msg) {
  symbol <- cli::col_green(cli::symbol$tick)
  msg <- cli::col_blue(paste(symbol, msg))
  message(msg)
}

```

Alternatively, maybe a global option such as `options(targets.default_text_color = "white")` could be nice?

Cheers,

**Comment by wlandau** [comment, marked as answer]
This is an instance of https://github.com/r-lib/cli/issues/467, so unfortunately user-generated messages will still be red. But for messages generated by `targets`, I think it makes sense to use red (`cli::col_red()`) for errors and warnings and the default theme color (`cli::col_none()`) everywhere else. Implemented in 3ff0ba6d29cb68e2963dadad0b7cba2050527b64. Still doesn't seem to fix what's going on in terminals outside RStudio though.

**Reply by adhi-r** [reply]
so I'm having the opposite issue - everything is white so it's hard to find the error. how can i fix this?
<img width="738" alt="image" src="https://github.com/ropensci/targets/assets/13878159/4cccb84c-d663-4701-81f9-b799b96853bb">

**Reply by wlandau** [reply]
Not all platforms support colors, e.g. if you are using a very simple terminal interface. You might have a look at https://cli.r-lib.org/reference/cli-config.html to see if there is an option or environment variable you can toggle in `_targets.R` to enable colors.
